modules/core/shared/src/main/scala/io/circe/JsonObject.scala

```diff
@@ -253,19 +252,20 @@ object JsonObject {
    * Construct a [[JsonObject]] from an [[scala.collection.Iterable]] (provided for optimization).
    */
   final def fromIterable(fields: Iterable[(String, Json)]): JsonObject = {
-    if (fields.isEmpty) {
-      empty
-    } else {
-      val map = new LinkedHashMap[String, Json]
-      val iterator = fields.iterator
-
-      while (iterator.hasNext) {
-        val (key, value) = iterator.next()
-
-        map.put(key, value)
-      }
-
-      fromLinkedHashMap(map)
+    val fieldsSeq = fields.iterator.toSeq
+    (fieldsSeq.size: @switch) match {
```

Comment:
I think you should avoid converting the `fields` into a seq, just to use `size`, use `sizeCompare` or `sizeIs` on the `Iterable` and determine if its 0 or 1, or greater instead... this way you don't have to materialize the Seq, and `sizeCompare` is cheap in comparison.